src/java/boa/compiler/visitors/TypeCheckingVisitor.java

```diff
@@ -937,6 +955,36 @@ public void visit(final VarDeclStatement n, final SymbolTable env) {
 
 			if (rhs != null && !lhs.assigns(rhs) && !env.hasCast(rhs, lhs))
 				throw new TypeCheckException(n.getInitializer(), "incorrect type '" + rhs + "' for assignment to '" + id + ": " + lhs + "'");
+			
+			if (n.getType() instanceof OutputType) {
+                BoaModel model = (BoaModel) SymbolTable.getMLAggregatorType(((OutputType) n.getType()).getId().getToken());
```

Comment:
why is this code here, instead of in `public void visit(final OutputType n, final SymbolTable env)` where the outputtype's are actually checked?

Reply:
I have moved the code into public void visit(final OutputType n, final SymbolTable env)